File: day02/part1.nim
import tables
import strutils 

var 
  twos: int
  threes: int

for line in "input".lines:
  var d = initTable[char,int]()
  for ch in line:
    d.mgetOrPut(ch, 0) += 1

  var 
    found_twos: bool
    found_threes: bool

  for k, v in d.pairs():
    if v == 2:
      found_twos = true
    if v == 3:
      found_threes = true

  if found_twos:
    twos += 1
  if found_threes:
    threes += 1

echo(twos * threes)

File: day02/part1.py
from collections import defaultdict

twos = 0
threes = 0
for line in open("input").readlines():
    line = line.strip()
    d = defaultdict(int)
    for ch in line:
        d[ch] += 1
    found_twos = False
    found_threes = False
    for k, v in d.items():
        if v == 2:
            found_twos = True
        if v == 3:
            found_threes = True
    twos += found_twos
    threes += found_threes
print(twos * threes)
